src/main/scala/cromwell/engine/backend/jes/authentication/JesAuthentication.scala

```diff
@@ -0,0 +1,66 @@
+package cromwell.engine.backend.jes.authentication
+
+import cromwell.engine.WorkflowDescriptor
+import cromwell.engine.backend.jes.{JesConfigurationSingleton, JesConfiguration, JesInterface}
+import spray.json.JsObject
+
+import scala.util.{Failure, Success}
+
+/**
+ * This makes JesConnection non visible when mixing a JesAuthentication trait
+ */
+trait JesConnection {
+  val JesConnection: JesInterface
+}
+
+/**
+ * Trait for JesAuthentication
+ */
+trait JesAuthentication { self: JesConfiguration with JesConnection =>
+
+  private def getRefreshToken(descriptor: WorkflowDescriptor) = descriptor.workflowOptions.get("refresh_token")
+  private def getAccessToken(descriptor: WorkflowDescriptor) = descriptor.workflowOptions.get("access_token")
+
+  def authenticated[A](f: JesInterface => A)(implicit descriptor: WorkflowDescriptor) = {
```

Comment:
Right, at first I thought I could take advantage of the implicit everywhere but then realized that in most cases the workflowDescriptor is "burried" in a BackendCall so I had to specify it.
Anyway like I said in an other comment I think I will remove the workflow parameter since it was only needed to retrieve the refreshToken from the options, and use that to adapt the credentials on the fly, but since we take the "Service Account only" path it's not required anymore.